src/api/stats.ts

```diff
@@ -130,70 +132,89 @@ export const convertToUTC = (
+            .order('catalog_name');
+
+        const today = new Date();
+
+        // TODO (locale) allow users to have proper locale settings used for start and end of weeks
+        // startOf/endOf functions can give some odd results so just forcing exactly
+        //  what days we want to say are the start and end of a week based on the
+        //  current day.
+        const weekStart = isSunday(today) ? today : previousSunday(today);
+        const weekEnd = isSaturday(today) ? today : nextSaturday(today);
+
+        switch (filter) {
+            // Day Range
+            case 'today':
+                queryBuilder = queryBuilder
+                    .eq('ts', convertToUTC(today, dailyGrain))
+                    .eq('grain', dailyGrain);
+                break;
+            case 'yesterday':
+                queryBuilder = queryBuilder
+                    .eq('ts', convertToUTC(subDays(today, 1), dailyGrain))
+                    .eq('grain', dailyGrain);
+                break;
+
+            // Week Range
+            case 'thisWeek':
+                queryBuilder = queryBuilder
+                    .gte('ts', convertToUTC(weekStart, dailyGrain))
+                    .lte('ts', convertToUTC(weekEnd, dailyGrain))
+                    .eq('grain', dailyGrain);
+                break;
+            case 'lastWeek':
+                queryBuilder = queryBuilder
+                    .gte('ts', convertToUTC(subWeeks(weekStart, 1), dailyGrain))
+                    .lte('ts', convertToUTC(subWeeks(weekEnd, 1), dailyGrain))
+                    .eq('grain', dailyGrain);
+                break;
+
+            // Month Range
+            case 'thisMonth':
+                queryBuilder = queryBuilder
+                    .eq('ts', convertToUTC(today, monthlyGrain))
+                    .eq('grain', monthlyGrain);
+
+                break;
+            case 'lastMonth':
+                queryBuilder = queryBuilder
+                    .eq('ts', convertToUTC(subMonths(today, 1), monthlyGrain))
+                    .eq('grain', monthlyGrain);
+                break;
+
+            default:
+                throw new Error('Unsupported filter used in Stats Query');
+        }
+
+        return queryBuilder;
+    };
+
+    // This could probably be written in a fancy functional-programming way with
+    // clever calls to concat and map and slice and stuff,
+    // but I want it to be dead obvious what's happening here.
+    while (index < names.length) {
+        // Have to do this to capture `index` correctly
+        const prom = promiseGenerator(index);
+        promises.push(limiter(() => prom));
+
+        index = index + CHUNK_SIZE;
     }
 
-    return supabaseRetry(() => queryBuilder, '').then(
-        handleSuccess<CatalogStats[]>,
-        handleFailure
-    );
+    const response = await Promise.all(promises);
+    const errors = response.filter((r) => r.error);
+    return errors[0] ?? { data: response.flatMap((r) => r.data) };
```

Comment:
This is one of the few places where we need to get all the data back so needing to map the response.